src/adrs/0129349218/README.md

```diff
@@ -0,0 +1,506 @@
+
+### Debug Info
+
+The generation of `debug_info` should be a controlled process, not an automatic default for every error.
+
+Avoid generating stack traces or other expensive debug information proactively. Instead, generate it only when it's determined to be necessary, for example, based on the error's severity or type. For user-facing errors like input validation (`INVALID_ARGUMENT`), `debug_info` is often unnecessary.
+
+### Batching and Wrapping Errors
+
+The specification supports scenarios where multiple errors need to be reported together or where an error needs to be wrapped with additional context. The `causes` field, which is an array of `UniversalError` objects, is used for this purpose. This is useful for:
+
+- **Batch operations**: Reporting all errors that occurred during a batch process in a single response.
+- **Error wrapping**: Adding context to an error from a downstream service.
+
+### Validation and Input Handling
+
+For input validation, it is critical to provide precise feedback about which fields are invalid. The `subject` field, which can be a JSON Pointer (`RFC 6901`), identifies the specific input field that caused the error.
+
+For example, given the following command payload:
+
+```js
+const command = {
+  data: { amount: '1000', currency: 'CUP' },
+};
+```
+
+An error response can pinpoint the invalid field:
+
+```js
+const error = {
+  subject: '/data',
+  info: {
+    domain: 'com.mybusiness.myapp',
+    reason: 'BAD_REQUEST',
+  },
+  code: 'INVALID_ARGUMENT',
+  causes: [
+    {
+      subject: '/currency',
+      info: {
+        domain: 'com.mybusiness.myapp',
+        reason: 'INVALID_CURRENCY',
+        metadata: { valid_currencies: 'USD' },
+      },
+      // ...
+    },
+  ],
+};
+```
+
+This structure allows a UI to display validation errors next to the corresponding input fields. When wrapping errors, the `subject` remains relative to its local context.
+
+## Resolution
+
+The structure of the Universal Error is defined as follows:
+
+```typescript
+type UniversalError = {
+  specversion: SpecVersion;
+  code: Code;
+  message: MessageTemplate;
+  info: ErrorInfo;
+  causes: UniversalError[];
+  visibility: Visibility;
+
+  //
+  // Optional Information
+  //
+  subject?: Subject;
+  id?: Id;
+  time?: Time;
+  help?: Help;
+  debug_info?: DebugInfo;
+  localized_message?: LocalizedMessage;
+  retry_info?: RetryInfo;
+  source_id?: SourceId;
+};
+
+/**
+ * An opaque value that helps pinpoint the location in the code where the error
+ * was constructed. While not guaranteed to be unique, it ideally provides a
+ * stable reference to the error's origin.
+ */
+type SourceId = string;
+
+/**
+ * The version of the specification that the error uses.
+ * Monotonically increasing version numbers are used to identify the evolution
+ * of the error format. The version number is represented as a non-negative
+ * integer.
+ */
+type SpecVersion = integer;
```

Comment:
```suggestion
type SpecVersion = number;
```

Depending on how wild you want to get, you could use a branded type (or similar alternative) for `PositiveInteger` so it's very clear with: `type SpecVersion = PositiveInteger`